packages/flutter/lib/src/widgets/autofill.dart

```diff
@@ -0,0 +1,182 @@
+///              TextField(
+///                controller: billingAddress1,
+///                autofillHints: <String>[AutofillHints.streetAddressLine1],
+///              ),
+///              TextField(
+///                controller: billingAddress2,
+///                autofillHints: <String>[AutofillHints.streetAddressLine2],
+///              ),
+///            ],
+///          ),
+///        ),
+///        const Text('Credit Card Information'),
+///        // The credit card number and the security code are grouped together as
+///        // some platforms are capable of autofilling both fields.
+///        AutofillGroup(
+///          child: Column(
+///            children: <Widget>[
+///              TextField(
+///                controller: creditCardNumber,
+///                autofillHints: <String>[AutofillHints.creditCardNumber],
+///              ),
+///              TextField(
+///                controller: creditCardSecurityCode,
+///                autofillHints: <String>[AutofillHints.creditCardSecurityCode],
+///              ),
+///            ],
+///          ),
+///        ),
+///        const Text('Contact Phone Number'),
+///        // The phone number field can still be autofilled despite lacking an
+///        // `AutofillScope`.
+///        TextField(
+///          controller: phoneNumber,
+///          autofillHints: <String>[AutofillHints.telephoneNumber],
+///        ),
+///      ],
+///    );
+///  }
+/// ```
+/// {@end-tool}
+class AutofillGroup extends StatefulWidget {
+  /// Creates a scope for autofillable input fields.
+  ///
+  /// The [child] argument must not be null.
+  const AutofillGroup({
+    Key key,
+    @required this.child,
+  }) : assert(child != null),
+       super(key: key);
+
+  /// Returns the closest [AutofillScope] which encloses the given context.
+  ///
+  /// In order to interact with the platform's autofill mechanism,
+  /// [AutofillTrigger]s need to call [AutofillScope.attach] on their closest
+  /// [AutofillScope], instead of calling [TextInputClient.attach].
+  static _AutofillGroupState of(BuildContext context) {
+    final _AutofillScope scope = context.dependOnInheritedWidgetOfExactType<_AutofillScope>();
+    return scope?._scope;
+  }
+
+  /// {@macro flutter.widgets.child}
+  final Widget child;
+
+  @override
+  _AutofillGroupState createState() => _AutofillGroupState();
+}
+
+class _AutofillGroupState extends State<AutofillGroup> with AutofillScopeMixin {
+  @override
+  Iterable<AutofillClient> get autofillClients {
+    final List<AutofillClient> clients = <AutofillClient>[];
+    void visit(Element element) {
+      if (element is AutofillScope)
+        return;
+      if (element is AutofillClient) {
+        clients.add(element as AutofillClient);
+      } else if (element is StatefulElement && element.state is AutofillClient) {
+        clients.add(element.state as AutofillClient);
+      } else {
+        element.visitChildElements(visit);
```

Comment:
If you have nested AutofillGroups, will this not traverse into the nested group? Is that expected?

Reply:
Yes, I think that can be used to exclude the inner group from the outer group in autofill. E.g., 
```dart
AutofillGroup(
  child: Column(
    children: <Widget>[
      usernameField,
      AutofillGroup(child: emailField),
      passwordField,
   ],
  ),
)
```

Reply:
Sorry, my comment was a little unclear. To me it seems the code _will_ traverse into the nested group. What stops the outer group from seeing the emailField in your example?

Reply:
```dart
      if (element is AutofillScope)
        return;
```
 Looks like I forgot the case where the state is an `AutofillScope`.

Reply:
Probably worth adding a test for this case.